### ISSUE TITLE
Use regular expression to search package descriptions for 3PC and 5PC

### ISSUE THREAD

**Issue by kbower** [kbower/tickettest1#899]
_Imported from trac ticket 899.
Created by **kbower**
Opened in trac: May 19, 2011, 3:45:31 PM
Last modified in trac: May 20, 2011, 10:07:59 AM_
***
**Implementation note: put compiled regular expression object as the value for the 'description' items**

Well, I don't want to use that because then a description such as
"25" CUPCAKE WALL MIRROR" and
"x CHAISE 46X63 TP CYBERSPC GEM"
would match.

I'll use a regular expression to fix this, but like I mentioned, it would probably faster if you did this manually with SQL in the meantime.

On 5/19/2011 3:30 PM, Wayne Liburd wrote:
>
> Kent,
>
> Can we make the rule check for Upper(‘3%PC’) or Upper(‘5%PC’) respectively?
>
> Thanks,
> Wayne Liburd
> From: Kent Bower [mailto:kb@retailarchitects.com]
> Sent: Thursday, May 19, 2011 3:19 PM
> To: Wayne Liburd
> Subject: Re: Illegal Ottoman substitution
>
> Wayne,
>     Note there are several packages defined with a space in the description, such as pasted below.
>
> These won't work with our package rules currently (even the UPOC or SECT ones).  I could fix this in the rules, but for times sake, it would be faster probably if you update these descriptions with SQL.
>
> SQL> select itm_cd, des, mnr_cd from itm where des like '%3 PC%' or des like '%5 PC%';
>
> ITM_CD    DES                            MNR_C
> --------- ------------------------------ -----
> 742032319 13 PC COFFEE SET/ PEACH SEVRES 740
> 743032310 13 PC COFFEE SET PURPLE SEVRES 740
> ROYA8     5 PC ENTERTAINMENT WALL        WALL
> BAYS5     3 PC BED 5/0                   HBFB
> BAYS6     3 PC BED 6/6                   HBFB
> REGS1     5 PC TABLE AND 4SC             DINF
> TARA5     3 PC DAYBED                    DAYB

**Comment by kbower** [comment]
_**kbower** trac comment on May 19, 2011, 4:17:35 PM:_

Well, I've noticed dashes as well as spaces and like you said, lower case, 5pc, etc.

But I've noticed our current rules will think 13PC is a 3PC according to the description.  So I think that I will get slightly more sophisticated and use a regular expression.

On 5/19/2011 4:11 PM, Wayne Liburd wrote:
Kent,
Since this would be a onetime update and is a natural way for anyone of the buyers or re-buyers to key the description let's check for '3PC', '5PC', '3 PC' and '5 PC' respectively.
Thanks,*
Wayne Liburd*
